picard/script/functions.py

```diff
@@ -1381,3 +1383,105 @@ def func_countryname(parser, country_code, translate=""):
     if translate:
         return gettext_countries(name)
     return name
+
+
+def _make_date(date_to_parse, date_order="ymd"):
+    """Create a datetime object from a date string with optional date order specified.
+
+    Args:
+        date_to_parse (str): Date string to parse
+        date_order (str, optional): Order of date elements. Can be "ymd", "mdy" or "dmy". Defaults to "ymd".
+
+    Returns:
+        datetime: Object derived from the parameters provided
+    """
+    date_order = date_order.lower()
+    if date_order not in {'ymd', 'mdy', 'dmy'}:
+        date_order = 'ymd'
+    dayfirst = date_order[0] == 'd'
+    yearfirst = date_order[0] == 'y'
```

Comment:
The code above use 3 tests, and sets `date_order` to `ymd` while it isn't used afterwards.

I think following code is equivalent with less tests and sets:

```python
if date_order == 'dmy':
    dayfirst, yearfirst = True, False
elif date_order == 'mdy':
    dayfirst, yearfirst = False, False
else:  # defaults to 'ymd'
    dayfirst, yearfirst = False, True
```

What do you think?